compiler/ballerina-lang/src/main/java/org/ballerinalang/compiler/plugins/CompilerPlugin.java

```diff
@@ -116,8 +116,25 @@ default void setCompilerContext(CompilerContext context) {
     /**
      * Notifies when the code generated phase is completed.
      *
-     * @param packageID packageId of the generated code
+     * @param packageID  packageId of the generated code
      * @param binaryPath path to the generated binary file (balx)
      */
     void codeGenerated(PackageID packageID, Path binaryPath);
+
+
+    /**
+     * Notifies when the compiler starts executing compiler plugins for a particular module.
+     *
+     * @param packageID PackageID of the module
+     */
+    default void pluginExecutionStarted(PackageID packageID) {
```

Comment:
Do we need the "plugin" prefix, given that these are methods in `CompilerPlugin`? Wouldn't just `executionStarted` and `executionCompleted` be enough?

Reply:
I thought the same, But we have a method called codeGenerated, which is executing in a later phase. So executionCompleted is not the right word, So that is why I have added plugin prefix.